pkg/netceptor/netceptor.go

```diff
@@ -784,20 +784,20 @@ func (s *Netceptor) monitorConnectionAging() {
 	for {
 		select {
 		case <-time.After(5 * time.Second):
-			timedOut := make([]context.CancelFunc, 0)
+			timedOut := make(map[string]context.CancelFunc, 0)
 			s.connLock.RLock()
-			for i := range s.connections {
-				conn := s.connections[i]
-				conn.lastReceivedLock.RLock()
-				if time.Since(conn.lastReceivedData) > s.maxConnectionIdleTime {
-					timedOut = append(timedOut, s.connections[i].CancelFunc)
+			for conn := range s.connections {
+				connInfo := s.connections[conn]
+				connInfo.lastReceivedLock.RLock()
+				if time.Since(connInfo.lastReceivedData) > s.maxConnectionIdleTime {
+					timedOut[conn] = s.connections[conn].CancelFunc
 				}
-				conn.lastReceivedLock.RUnlock()
+				connInfo.lastReceivedLock.RUnlock()
 			}
 			s.connLock.RUnlock()
-			for i := range timedOut {
-				s.Logger.Warning("Timing out connection, idle for the past %s\n", s.maxConnectionIdleTime)
-				timedOut[i]()
+			for conn := range timedOut {
+				s.Logger.Warning("Timing out connection from %s to %s, idle for the past %s\n", s.nodeID, conn, s.maxConnectionIdleTime)
```

Comment:
I think the `from nodeA to nodeB` implies connection direction. Meaning I interpret that as nodeA is dialing a tcp listener on nodeB. However, it could be that nodeB dialed nodeA.

I think we should drop `s.nodeID` and just say "timing out connection %s, idle for the past %s`